lms/djangoapps/certificates/api.py

```diff
@@ -273,6 +274,27 @@ def set_cert_generation_enabled(course_key, is_enabled):
         log.info(u"Disabled self-generated certificates for course '%s'.", unicode(course_key))
 
 
+def is_certificate_invalid(student, course_key):
+    """Check that whether the student in the course has been invalidated
+    for receiving certificates.
+
+    Arguments:
+        student (user object): logged-in user
+        course_key (CourseKey): The course identifier.
+
+    Returns:
+        Boolean denoting whether the student in the course is invalidated
+        to receive certificates
+    """
+    is_invalid = False
+    certificate = GeneratedCertificate.certificate_for_student(student, course_key)
+    if certificate and certificate.status == CertificateStatuses.unavailable:
+        # in case of status 'un-available', check that if this certificate is invalidated
```

Comment:
@schenedx in case of invalidation cert status become un-available. That's why I am checking only for this. [invalidate](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/certificates/models.py#L284)